standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -1394,9 +1394,32 @@ List<ColumnStatistics> getPartitionColumnStatistics(
    * @throws InvalidObjectException error dropping the stats
    * @throws InvalidInputException bad input, such as null table or database name.
    */
-  boolean deletePartitionColumnStatistics(String catName, String dbName, String tableName,
-      String partName, List<String> partVals, String colName, String engine)
-      throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException;
+  default boolean deletePartitionColumnStatistics(String catName, String dbName, String tableName,
+    String partName, List<String> partVals, String colName, String engine)
+    throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException{
+    return false;
+  };
+
+  /**
+   * Deletes column statistics if present associated with a given db, table, partition and a list of cols. If
+   * null is passed instead of a colName, stats when present for all columns associated
+   * with a given db, table and partition are deleted.
+   * @param catName catalog name.
+   * @param dbName database name.
+   * @param tableName table name.
+   * @param partName partition name.
+   * @param partVals partition values.
+   * @param colNames a list of column names.
+   * @param engine engine for which we want to delete statistics
+   * @return Boolean indicating the outcome of the operation
+   * @throws NoSuchObjectException no such partition
+   * @throws MetaException error access the RDBMS
+   * @throws InvalidObjectException error dropping the stats
+   * @throws InvalidInputException bad input, such as null table or database name.
+   */
+  boolean deletePartitionMultiColumnStatistics(String catName, String dbName, String tableName,
+    String partName, List<String> partVals, List<String> colNames, String engine)
```

Comment:
the `partVals` never be used?

Reply:
It's used around ObjectStore.java Line 10266